### PR TITLE
Implement distro selection UI

### PR DESCRIPTION
**Pull request by SirCmpwn**

### CODE REVIEW

**Comment by eatonphil** [comment]
Could you left-align the tiles like the screenshot in #37?

**Comment by SirCmpwn** [comment]
I prefer them like this. Is it important to you?

**Comment by eatonphil** [comment]
Yes, I think they should be left-aligned.

File: src/linodes/components/DistroVendor.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      <div
+        onClick={() => onClick(selectedVersion)}
+        className={`distro ${
+          vendor.versions.find(v => v.id === selected) ? 'selected' : ''
+        }`}
+      >
+        <header className={`dropdown ${open ? 'open' : ''}`}>
+          <h3>
+            {label(selectedVersion.label)}
+            <button
```
Comment (eatonphil) [inline src/linodes/components/DistroVendor.js]: Why aren't you using the dropdown component?

Comment (SirCmpwn) [inline src/linodes/components/DistroVendor.js]: They don't behave like other dropdowns, and shouldn't. They also need more subtle layout and style applied to them.

**Comment by SirCmpwn** [comment]
Adding some tests I missed to catch up on coverage.

**Comment by SirCmpwn** [comment]
Ready for further review + merge.

File: src/linodes/components/SourceSelection.js [hunk 2]
```diff
@@ -1,24 +1,54 @@
...
+            selected={source}
+            vendor={v}
+            key={v.name}
+            onClick={s => onSourceSelected(s)}
+          />)}
+      </div>
+    );
   }
 
   renderSourceTabs() {
```
Comment (eatonphil) [inline src/linodes/components/SourceSelection.js]: Any reason not to be using `renderTabs` from LinodeDetailPage?

Comment (eatonphil) [inline src/linodes/components/SourceSelection.js]: In addition to just doing this for you, it also already has tests so we don't need to be duplicating those.

Comment (eatonphil) [inline src/linodes/components/SourceSelection.js]: Ah right, that doesn't exist yet. Nevermind, can refactor later.

Comment (eatonphil) [inline src/linodes/components/SourceSelection.js]: (I refactored tabs and their tests in the settings alert PR)

Comment (SirCmpwn) [inline src/linodes/components/SourceSelection.js]: 1. The linode detail page tabs are heavily invested in routing, which is not involved on this page (nor do I wish to involve it)
2. A component depending on a layout makes me squeamish

Comment (eatonphil) [inline src/linodes/components/SourceSelection.js]: I guess we'll talk about it in my PR

Comment (SirCmpwn) [inline src/linodes/components/SourceSelection.js]: Ah, feel free to file another GH issue for refactoring this later.

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**